hoomd/md/integrate.py

```diff
@@ -278,38 +279,42 @@ class Integrator(_DynamicIntegrator):
             perform computations during the half-step of the integration.
     """
 
-    def __init__(self,
-                 dt,
-                 integrate_rotational_dof=False,
-                 forces=None,
-                 constraints=None,
-                 methods=None,
-                 rigid=None,
-                 half_step_hook=None):
-
+    def __init__(
+        self,
+        dt,
+        integrate_rotational_dof=False,
+        forces=None,
+        constraints=None,
+        methods=None,
+        rigid=None,
+        half_step_hook=None,
+    ):
         super().__init__(forces, constraints, methods, rigid)
 
         self._param_dict.update(
             ParameterDict(
                 dt=float(dt),
                 integrate_rotational_dof=bool(integrate_rotational_dof),
                 half_step_hook=OnlyTypes(hoomd.md.HalfStepHook,
-                                         allow_none=True)))
+                                         allow_none=True),
+            ))
 
         self.half_step_hook = half_step_hook
 
     def _attach_hook(self):
         # initialize the reflected c++ class
-        self._cpp_obj = _md.IntegratorTwoStep(
-            self._simulation.state._cpp_sys_def, self.dt)
+        # checking for None allows deriving classes to construct this first
+        if self._cpp_obj is None:
+            self._cpp_obj = _md.IntegratorTwoStep(
+                self._simulation.state._cpp_sys_def, self.dt)
```

Comment:
Just wanted to add a note because this was a spot I wanted someone to have a look at.

Without this change, I think the `_cpp_obj` created in `mpcd.Integrator` would get overwritten by this one?

My alternative was to leave this alone & call the grandparent method from `mpcd.Integrator`. This would work currently because this method doesn't do anything other than initialize `_cpp_obj` then call the parent method, but I was worried that I would need to duplicate anything that got added here in future.

Reply:
I would also call the grandparent method. I don't anticipate adding more logic to this integrator attach hook.